### PR TITLE
NEWRELIC-6211: (winston) also instrument winston.loggers.add

### PR DESCRIPTION
**Pull request by jordigh**
There is a different part of the Winston API that our current instrumentation simply misses. It's possible to create a logger by directly calling `createLogger`, but it's also possible to add new loggers to the list of loggers by doing `winston.loggers.add` with almost the same syntax.

This factors out a bit of the logic so that both methods of creating winston loggers are instrumented.

## Proposed Release Notes

* Instrumented `winston.loggers.add` so it works like `winston.createLogger`.

## Links

* Closes NEWRELIC-6211

### CODE REVIEW

**Review by jmartin4563 — CHANGES_REQUESTED** [review]
Comment about the supportability metric move

File: lib/instrumentation/winston.js [hunk 1]
```diff
@@ -26,29 +26,63 @@ module.exports = function instrument(agent, winston, _, shim) {
     return
   }
 
+  createModuleUsageMetric('winston', agent.metrics)
```
[the remark this answers is not in the document]
Comment (jordigh) [reply lib/instrumentation/winston.js:29]: Right, so I reasoned that the real instrumentation was whenever `createLogger`/`loggers.add` was instrumented, i.e. during `require`, whether it was called with stream options or not. I deliberately created a test that if you created two loggers, one with each method, we would only record a single instrumentation.

Do you think that's the wrong idea? If for some reason someone calls both `createLogger` and `loggers.add`, should we count that as two instrumentations in our metrics?

[the remark this answers is not in the document]
Comment (bizob2828) [reply lib/instrumentation/winston.js:29]: Yea I think this has to move back into instrumentation.  Per the spec

> Supportability/Logging/{language}/{framework}/{enabled\|disabled}`|**MUST** be generated by agents when instrumentation is installed for application logging. `disabled` **SHOULD*      * be generated if possible, although some agents' implementation may preclude it. This metric does not take the enabled/disabled state of individual features (i.e. metrics, forward      ing) into account. It is expected to be sent once per agent connect (not per harvest). `framework` names are at the discretion of the language agent teams.

[the remark this answers is not in the document]
Comment (jordigh) [reply lib/instrumentation/winston.js]: Okay, changed.

**Review by jmartin4563 — APPROVED** [review]

**Pull request merged**